extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLAbstractHandler.java

```diff
@@ -58,9 +58,16 @@ public void handle(final RoutingContext ctx) {
             currentManagedContext.activate();
             handleWithIdentity(ctx);
 
-            ctx.response().bodyEndHandler((e) -> {
-                currentManagedContext.terminate();
-            });
+            ctx.response()
+                    .endHandler((e) -> {
+                        currentManagedContext.terminate();
+                    })
+                    .exceptionHandler((e) -> {
+                        currentManagedContext.terminate();
+                    })
+                    .closeHandler((e) -> {
+                        currentManagedContext.terminate();
+                    });
```

Comment:
Note that I'm not familiar at all with this. But I'm a bit worried you have to specifically declare these for every response.

Also what happens if `handleWithIdentity()` fails given you already have activated the context before calling the method?

Reply:
Also, while we're at it, I would avoid creating 3 lambdas for this given it's runtime code.